doc/guides/dts/index.rst

```diff
@@ -346,44 +346,534 @@ Below is sample header content generated for this node, in
+generates these macros:
+
+.. code-block:: c
+
+   #define DT_NXP_KINETIS_SIM_40047000_FOO_BASE_ADDRESS    0x40047000
+   #define DT_NXP_KINETIS_SIM_40047000_FOO_SIZE            4192
+
+Interrupt-related macros
+------------------------
+
+Take these devicetree nodes as an example:
+
+.. code-block:: none
+
+   timer@123 {
+   	compatible = "vendor,timer";
+   	interrupts = <1 5 2 6>;
+   	interrupt-parent = <&intc>;
    };
 
-This would generate additional ``DT_ALIAS_I2C_1_...`` aliases for all
-properties in the output for the node with the devicetree label ``i2c``.
+   intc: interrupt-controller {
+   	compatible = "vendor,intc";
+   };
+
+Assume that the binding for the interrupt controller (which would have
+``compatible: "vendor,intc"``) has this:
+
+.. code-block:: yaml
+
+   interrupt-cells:
+       - irq
+       - priority
+
+Then these macros get generated:
+
+.. code-block:: c
 
-Aliases that replace the property name part can also be generated, e.g. via
-``*-names = "foo", "bar"`` properties. For example, ``reg-names = "control",
-"mem"`` will generate ``DT_<node>_CONTROL_BASE_ADDRESS/SIZE`` and
-``DT_<node>_MEM_BASE_ADDRESS_SIZE`` aliases.
+   #define DT_VENDOR_TIMER_123_IRQ_0                   1
+   #define DT_VENDOR_TIMER_123_IRQ_0_PRIORITY          5
+   #define DT_VENDOR_TIMER_123_IRQ_1                   2
+   #define DT_VENDOR_TIMER_123_IRQ_1_PRIORITY          6
+
+These macros have the the format ``DT_<node>_IRQ_<index>(_<name>)``, where
+``<node>`` is the node prefix, ``<index>`` is an index that identifies the
+particular interrupt, and ``<name>`` is the identifier for the cell value (a
+number within ``interrupts = <...>``), taken from the binding.
+
+Bindings for interrupt controllers are expected to declare a cell named ``irq``
+in ``interrupt-cells``, giving the interrupt number. The ``_<name>`` suffix is
+skipped for macros generated from ``irq`` cells, which is why there's e.g. a
+``DT_VENDOR_TIMER_123_IRQ_0`` macro and no ``DT_VENDOR_TIMER_123_IRQ_0_IRQ``
+macro.
+
+If the interrupt controller in turn generates other interrupts, a multi-level
+interrupt encoding is used for the interrupt number. See
+:ref:`multi_level_interrupts` for more information.
+
+Additional macros that use names instead of indices for interrupts can be
+generated by including an ``interrupt-names`` property on the
+interrupt-generating node. For example, ``interrupt-names = "timer-a",
+"timer-b"`` gives these extra macros:
+
+.. code-block:: c
+
+   #define DT_VENDOR_TIMER_123_IRQ_TIMER_A             1
+   #define DT_VENDOR_TIMER_123_IRQ_TIMER_A_PRIORITY    5
+   #define DT_VENDOR_TIMER_123_IRQ_TIMER_B             2
+   #define DT_VENDOR_TIMER_123_IRQ_TIMER_B_PRIORITY    6
+
+Values for properties generated from ``clocks``
+-----------------------------------------------
+
+Clocks work the same as other ``phandle-array`` properties, except the
+generated macros have ``CLOCK`` in them instead of ``CLOCKS``, giving for
```

Comment:
what about the actual source in the `.dts` files? is it `clock` or `clocks`? we should probably state it here.

Reply:
Its `clocks` which I take from the section header

Reply:
Try to make that more explicit with the Clocks work the same -> ``clocks`` work the same ...